compiler/test/stdlib/path.test.gr

```diff
@@ -0,0 +1,165 @@
+import Path from "path"
+import Result from "result"
+import Option from "option"
+import List from "list"
+
+// test that a variety of properties hold for valid relative/absolute paths
+let relTests = [
+  (
+    "./dir/file.txt",
+    "./dir",
+    "./dir/file.txt",
+    Some("file.txt"),
+    Some("file"),
+    Some(".txt"),
+  ),
+  ("file", ".", "./file", Some("file"), Some("file"), None),
+  ("dir/../../", "../..", "..", None, None, None),
+  (".git", ".", "./.git", Some(".git"), Some(".git"), None),
+  (".", "..", ".", None, None, None),
+  ("..", "../..", "..", None, None, None),
+  ("../../file", "../..", "../../file", Some("file"), Some("file"), None),
+  ("./", "..", ".", None, None, None),
+  (
+    "dir//file.txt///",
+    "dir",
+    "./dir/file.txt",
+    Some("file.txt"),
+    Some("file"),
+    Some(".txt"),
+  ),
+  (
+    ".\\dir\\file.txt",
+    ".",
+    "./.\\dir\\file.txt",
+    Some(".\\dir\\file.txt"),
+    Some(".\\dir\\file"),
+    Some(".txt"),
+  ),
+]
+
+List.forEach(((pathStr, expParent, expStr, expName, expStem, expExt)) => {
+  let path = Option.unwrap(Path.relative(pathStr))
+  assert Path.toString(path, Path.Posix) == expStr
+  assert Option.unwrap(Path.relative(expParent)) == Path.parent(path)
+  assert expName == Path.name(path)
+  assert expStem == Path.stem(path)
+  assert expExt == Path.extension(path)
+}, relTests)
+
+let absTests = [
+  ("/usr/bin", "/usr", "/usr/bin", Path.Root, Some("bin"), Some("bin"), None),
+  ("/bin", "/", "/bin", Path.Root, Some("bin"), Some("bin"), None),
+  ("/bin/dir/..", "/", "/bin", Path.Root, Some("bin"), Some("bin"), None),
+  ("/../..", "/", "/", Path.Root, None, None, None),
+  ("/", "/", "/", Path.Root, None, None, None),
+  ("C:/", "C:/", "C:/", Path.Drive("C"), None, None, None),
+  ("C:/.././..", "C:/", "C:/", Path.Drive("C"), None, None, None),
+  (
+    "C:/Users/me.txt",
+    "C:/Users",
+    "C:/Users/me.txt",
+    Path.Drive("C"),
+    Some("me.txt"),
+    Some("me"),
+    Some(".txt"),
+  ),
+  (
+    "label:/me",
+    "label:/",
+    "label:/me",
+    Path.Drive("label"),
```

Comment:
Ah yeah I guess drive labels do have to be one letter, good point. I can also change the implementation to use a `Char` instead of a `String` for getting the drive label.